src/functions.jl

```diff
@@ -343,6 +343,11 @@ for (Ts, Ss) in [(Rlist, Rlist), (Rlist, Olist), (Olist, Rlist)]
 
     end
 end
+
+# Specific methods to resolve ambiguities in 1.6
+Base.:*(x::Adjoint{<:Any, <:AbstractMatrix{T}}, y::KeyedVector{S}) where {T, S<:Number} = matmul(x,y)
```

Comment:
Why do you need the `where` clause? Can't the signature be simplified to:

```suggestion
Base.:*(x::Adjoint{<:Any, <:AbstractMatrix}, y::KeyedVector{<:Number}) = matmul(x, y)
```
Same goes below

Reply:
I know there are some weird cases where Base has <: Number, since 1.5 IIRC, and so you have to match this.

Reply:
>I know there are some weird cases where Base has <: Number, since 1.5 IIRC, and so you have to match this.

Yeah there's an edge case with subtyping `Number` specifically so it needs to be added in otherwise we get:
```julia
  Expression: axiskeys(A3 * V, 1) == Base.OneTo(3)
  MethodError: *(::Adjoint{Float64, Matrix{Float64}}, ::NamedDimsArray{(:v,), Int64, 1, KeyedArray{Int64, 1, Vector{Int64}, Base.RefValue{StepRange{Int64, Int64}}}}) is ambiguous. Candidates:
    *(a::Adjoint{var"#s16", var"#s17"} where {T1, var"#s16", var"#s17"<:AbstractMatrix{T1}}, b::NamedDimsArray{var"#s18", var"#s19", 1, A} where {var"#s18", var"#s19", A<:AbstractVector{var"#s19"}}) in NamedDims at /Users/glenn/.julia/packages/NamedDims/ty09T/src/functions_math.jl:77
    *(adjA::Adjoint{var"#s814", var"#s813"} where {var"#s814", var"#s813"<:AbstractMatrix{T}}, x::AbstractVector{S}) where {T, S} in LinearAlgebra at /Applications/Julia-1.6.app/Contents/Resources/julia/share/julia/stdlib/v1.6/LinearAlgebra/src/matmul.jl:110
    *(x::Adjoint{var"#s49", var"#s7"} where {var"#s49", var"#s7"<:(AbstractMatrix{T} where T)}, y::NamedDimsArray{L, T, 1, var"#s7"} where {L, T<:Number, var"#s7"<:(KeyedArray{T, 1, AT, KT} where {AT, KT})}) in AxisKeys at /Users/glenn/.julia/dev/AxisKeys/src/functions.jl:340
    *(x::AbstractMatrix{T} where T, y::NamedDimsArray{L, T, 1, var"#s7"} where {L, T<:Number, var"#s7"<:(KeyedArray{T, 1, AT, KT} where {AT, KT})}) in AxisKeys at /Users/glenn/.julia/dev/AxisKeys/src/functions.jl:340
    *(x::Adjoint{var"#s49", var"#s7"} where {var"#s49", var"#s7"<:(AbstractMatrix{T} where T)}, y::NamedDimsArray{L, T, 1, var"#s7"} where {L, T, var"#s7"<:(KeyedArray{T, 1, AT, KT} where {AT, KT})}) in AxisKeys at /Users/glenn/.julia/dev/AxisKeys/src/functions.jl:340
    *(x::AbstractMatrix{T} where T, y::NamedDimsArray{L, T, 1, var"#s7"} where {L, T, var"#s7"<:(KeyedArray{T, 1, AT, KT} where {AT, KT})}) in AxisKeys at /Users/glenn/.julia/dev/AxisKeys/src/functions.jl:340
  Possible fix, define
    *(::Adjoint{var"#s16", var"#s17"} where {T1, var"#s16", var"#s17"<:AbstractMatrix{T1}}, ::NamedDimsArray{L, T, 1, var"#s7"} where {L, T<:Number, var"#s7"<:(KeyedArray{T, 1, AT, KT} where {AT, KT})})
```